### ISSUE TITLE
Bug when doing --update-when-exists

### ISSUE THREAD

**Issue by mnowak-umich** [vildanbina/laravel-model-json#1]
I think there is an error using the update-when-exists option.

```Now, we are going to import your model (App\Models\Message) data from a JSON format.

 0/1 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%
   TypeError 

  array_intersect_key(): Argument #1 ($array) must be of type array, string given

  at vendor/laravel/framework/src/Illuminate/Collections/Arr.php:495
    491▕      */
    492▕     public static function only($array, $keys)
    493▕     {
    494▕         Log::info($array);
  ➜ 495▕         return array_intersect_key($array, array_flip((array) $keys));
    496▕     }
    497▕ 
    498▕     /**
    499▕      * Pluck an array of values from an array.

      +2 vendor frames 
  3   [internal]:0
      Vildanbina\ModelJson\Services\ImportService::Vildanbina\ModelJson\Services\{closure}()

      +16 vendor frames 
  20  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()
```

I think the error is in ImportService.php, starting at line 82. In particular, line 83 is replacing the item array with a single string (usually "id") and that causes line 86 to fail.

**Comment by mnowak-umich** [comment]
If I comment out line 83, the import works.

**Comment by vildanbina** [comment]
We had an issue because we had accidentally named two variables with the same name, and now this problem has been resolved in the latest release [v1.1](https://github.com/vildanbina/laravel-model-json/releases/tag/v1.1). 

Thank you for you contribute!

**Comment by mnowak-umich** [comment]
Awesome, thank you!

-- Mike Nowak, Software Engineer Staff Specialist

Center for Health Communications Research
Rogel Cancer Center | Michigan Medicine

From: Vildan Bina ***@***.***>
Date: Monday, March 20, 2023 at 08:03
To: vildanbina/laravel-model-json ***@***.***>
Cc: mnowak-umich ***@***.***>, Author ***@***.***>
Subject: Re: [vildanbina/laravel-model-json] Bug when doing --update-when-exists (Issue #1)
External Email - Use Caution

We had an issue because we had accidentally named two variables with the same name, and now this problem has been resolved in the latest release v1.1<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fvildanbina%2Flaravel-model-json%2Freleases%2Ftag%2Fv1.1&data=05%7C01%7Cmnowak%40med.umich.edu%7C218d7b214fc04ec1046208db293b27d1%7C1f41d613d3a14ead918d2a25b10de330%7C0%7C0%7C638149106264384003%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=9kGmItSH69nhkcjzZP065AUouGMISt1qCKy4345xUVo%3D&reserved=0>.

Thank you for you contribute!

—
Reply to this email directly, view it on GitHub<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fvildanbina%2Flaravel-model-json%2Fissues%2F1%23issuecomment-1476098974&data=05%7C01%7Cmnowak%40med.umich.edu%7C218d7b214fc04ec1046208db293b27d1%7C1f41d613d3a14ead918d2a25b10de330%7C0%7C0%7C638149106264384003%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=GrzdLBB4ZZMn%2Bqa2uwCUcAOJGoCrqUcsfC3QoEfHEVE%3D&reserved=0>, or unsubscribe<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAMB57BDM2O77CDI4TYTK33W5BBLFANCNFSM6AAAAAAV25FPFY&data=05%7C01%7Cmnowak%40med.umich.edu%7C218d7b214fc04ec1046208db293b27d1%7C1f41d613d3a14ead918d2a25b10de330%7C0%7C0%7C638149106264384003%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=VxjClZt%2FX8T2Vfgf%2FJn6%2FDMgTuYMgfp1Kb3BNntc9n8%3D&reserved=0>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
**********************************************************
Electronic Mail is not secure, may not be read every day, and should not be used for urgent or sensitive issues
